### DISCUSSION TITLE
Zoom does not work on video

### DISCUSSION THREAD

**Discussion by makovez** [immich-app/immich#12034] [Feature Request]
### The bug

When the video is stopped I can't exit from play mode to only show the frame inage. And I can't zoom the frame.

### The OS that Immich Server is running on

Android

### Version of Immich Server

V1.112.1

### Version of Immich Mobile App

1.112.1

### Platform with the issue

- [X] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
N/a
```

### Your .env content

```Shell
N/a
```

### Reproduction steps

```bash
1. Open Android app and play a video
2. Stop the video
3. Zoom in frame
...
```

**Comment by mmomjian** [comment]
I don’t believe this is a bug, we don’t support zooming into video. I guess it could be a feature request.

**Reply by 1661410107114145152525** [reply]
Absolutely @mmomjian . In general the video player would benefit from a bit more love in terms of controls, but yes, zooming is by far the most important one and nearly a standard in most modern gallery apps.

Same goes for the in my opinion, would be great to have some more possibilities here.

**Comment by baco7** [comment]
I agree it should be a feature. It works fine in google photos.

**Comment by rpimeekle** [comment]
I, for one, would love to see this feature added.

**Comment by regnalbokaj** [comment]
I think this feature would be great!
Any plans for this yet?

**Comment by rrvsh** [comment]
I'd like to work on adding this feature, but am unfamiliar with the codebase. if any contributors could point me in the right direction for the video viewport code that would be amazing - also, I've never used any other self hosted photos app so if anyone is aware of another one that has this feature please let me know so i can refer to their implementation

**Reply by bo0tzz** [reply]
I'm not all that familiar with the mobile code but I believe it's at https://github.com/immich-app/immich/blob/main/mobile/lib/pages/common/native_video_viewer.page.dart.

**Reply by rrvsh** [reply]
Thank you!

**Reply by makovez** [reply]
If you can do it, that would be really helpfull @rrvsh looking forward to that 💪

**Reply by regnalbokaj** [reply]
Also zooming on Live Photos would be great.
Might be related to video zoom. Maybe it can be checked, when implementing zoom for videos! 😃

**Comment by opbod** [comment]
Related: https://github.com/immich-app/immich/pull/22036

**Comment by 1661410107114145152525** [comment]
@rrvsh Any news on that? I really miss this feature.

@makovez Could you please check "web" too for this festure in the description? There are so many use cases. And also update the description that it is fitting the purpose of a feature request more? @bo0tzz maybe you could support if that account receives no notifications? (I have no edit rights, otherwise I wouldn't ask).

@bo0tzz Thanks for keeping everything clean and tidy, found this issue linked in another one. :)

**Comment by 1661410107114145152525** [comment]
@bo0tzz Might such stuff be possible with plugins in the future? Like in Android gallery apps where you can set a different app as the default video playery but in a bit more integrated way?

**Comment by 1661410107114145152525** [comment]
>1. Open Android app and play a video
> 2. *Stop the video* → ⚠️
> 3. Zoom in frame
...

I don't see why I need to stop the video to zoom in. Samsung Gallery and many others are able to do it without pausing. This gives the possibility to track further away things like e.g. birds while they are flying. Please use sth. like that as a reference. The issue description is rather a bug than an appropriate feature description.

**Comment by 1661410107114145152525** [comment]
As this got quite some likes, duplicate requests and is 1½ years old and is something that wouls really benefit at so many places as it affects the core functionality of Immich: Is there any chance this becomes discussed in the Immich team? @bo0tzz (not enforcing anything I know there are many, many issues etc., just asking politely, wearing the hopes of all the others here).

**Comment by opbod** [comment]
@1661410107114145152525 this is being discussed by the Immich team. The implementation on the table today does seem to allow video zooming while playing remains active (see https://github.com/immich-app/immich/pull/22036 -> screenshot/demo).

There were some comments few days ago and @goalie2002 is so kind to continue the work on this soon. 🤞 

Yesterday evening when hanging out with a (non-technical) friend, we were browsing photos and videos and he wanted to zoom in on a video just like The-App-That-Shall-Not-Be-Named supports. It's a natural thing to do/try/expect.

**Reply by 1661410107114145152525** [reply]
This is really great to hear! 😍 Thank you so much for the update! ❤️ @opbod

**Reply by 1661410107114145152525** [reply]
And a special thanks to @goalie2002 for working on that essential feature 🏆!

**Reply by 1661410107114145152525** [reply]
I just saw that the linked PR is about mobile only, according tonthe labels right? What about the web? Especially when using the web version on a mobile device but also on smaller TVs etc. this would be really helpful too. Are there any pleans on that too? @opbod

**Comment by mmomjian** [comment]
@1661410107114145152525 please stop spam replying and tagging people. This is not an efficient use of our GitHub repo.

**Reply by 1661410107114145152525** [reply]
@mmomjian This was at no point my intention, sorry if anyone got messages they did not wanted. Regarding the message where I replied to your comment, I get if you see it as not helpful, though I wanted to say that propably more than zooming in might be helpful and appreciate that it turned into a festure request). I am not an Github expert as I am no developer.

Besides that one reply, I mainly replied with one of the following reasons, I would not consider spam replying:

1. suggesting new ideas and perspectives on the topic
2. asking the ones who can edit the description to update it accordingly (that's what I call efficient btw, up to date descriptions, so nobody has to read through dozens of duplicate issues or threads) 
3. saying thanks in replies to people that tagged me in their reply and helped me a lot with their information or effort (I don't think one should stop showing gratitude for reasons of efficiency)

If any of that is unwanted in an issue type that is called discussion where everyone is free to subscribe or unsubscribe, please say so (if possible with an explanation why and where I can write such things instead) and I will keep it in mind, stop posting such stuff and help efficiency by reporting others doing the same things in the future. If not, it would be great to see the spam mark removed on my messages above. 

Thanks for reading through this. Took me quite a while as I am typing on my smartphone (+ it cleared everything due to battery issues), but I don't want to make the impression of spamming and resolve possible misunderstandings constructively and politely. :)

**Comment by overwatch3560** [comment]
What's the latest update on this is it possible to implement the feature or coming soon??

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by evanbarke** [comment]
I have prepared a PR to enable this https://github.com/immich-app/immich/pull/26586
